docs/conf.py

```diff
@@ -155,3 +157,41 @@
 # ----------------------------------------------------------------------------------
 
 plot_html_show_formats = False
+
+
+# Need this method for the sphinx.ext.linkcode extension
+def linkcode_resolve(domain, info):
+    """
+    Determine the URL corresponding to Python object
+    """
+    if domain != "py":
+        return None
+
+    module = sys.modules.get(info["module"])
+    if module is None:
+        return None
+
+    obj = module
+    for part in info["fullname"].split("."):
+        obj = getattr(obj, part)
+        is_valid_code_object = (
+            inspect.isclass(obj) or inspect.ismethod(obj) or inspect.isfunction(obj)
+        )
+        if not is_valid_code_object:
+            return None
+
+    full_file_name = inspect.getsourcefile(obj)
+    if full_file_name is None:
+        return None
+    repo_root = PurePath(__file__).parent.parent
+    file_name = PurePath(full_file_name).relative_to(repo_root)
+
+    try:
+        source, lineno = inspect.getsourcelines(obj)
+    except OSError:
+        linespec = ""
+    else:
+        ending_lineno = lineno + len(source) - 1
+        linespec = f"#L{lineno}-L{ending_lineno}"
+
+    return f"https://github.com/Qiskit/qiskit_sphinx_theme/tree/{release}/{file_name}/{linespec}"
```

Comment:
1. Needs to be Qiskit
2. The `/` between the file and linespec is not necessary
3. Using the release is wrong; in that case, it would be `stable/{version}`. But we still need to figure out how we're going to detect `main` vs branch. In the meantime, let's set the `branch_name` variable. We'll fill in the other edge cases soon.

```suggestion
    git_branch = "main"

    return f"https://github.com/Qiskit/qiskit/tree/{git_branch}/{file_name}{linespec}"
```